scripts/ci/dockerfile-assets/bash/extract-environment.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#
+# Provide environment variables for static frontends by checking referenced
+# variables in the code and inject them into index.html
+#
+
+set -e
+
+work_dir="/usr/share/nginx/html"
+file="$work_dir/index.html"
+
+placeholder="<!-- environment placeholder -->"
+
+function extract_environment() {
+  env_prefix="SI_PUBLIC_"
+  environment="{}"
+  env_names=$(grep -ohr "$env_prefix\w*" "$work_dir/" | sort | uniq)
```

Comment:
Bash question. If grep finds nothing when I run this command on my machine, it exits with an error status code. Will that crash the whole script? Maybe that's fine?

Reply:
Because we have `set -e` any command that exists with an error should crash the script.

The command should not exit with an error though, even if it does not find any references.
I have tried this on my linux and also in the Docker container that runs this script so we should be fine. Not sure if this is a mac thing.